Client/Frontend/Browser/BrowserPrompts.swift

```diff
@@ -41,39 +42,57 @@ struct MessageAlert: JSAlertInfo {
     let message: String
     let frame: WKFrameInfo
     let completionHandler: () -> Void
+    var shouldCallCompletion: Bool
+
+    init(message: String,
+         frame: WKFrameInfo,
+         completionHandler: @escaping () -> Void,
+         shouldCallCompletion: Bool) {
+        self.message = message
+        self.frame = frame
+        self.completionHandler = completionHandler
+        self.shouldCallCompletion = shouldCallCompletion
+    }
 
     func alertController() -> JSPromptAlertController {
         let alertController = JSPromptAlertController(
             title: titleForJavaScriptPanelInitiatedByFrame(frame),
             message: message,
             preferredStyle: .alert)
         alertController.addAction(UIAlertAction(title: .OKString, style: .default) { _ in
+            guard shouldCallCompletion else { return }
+
             self.completionHandler()
         })
         alertController.alertInfo = self
         return alertController
     }
-
-    func cancel() {
-        completionHandler()
-    }
+    // This alert doesn't have a cancel action
+    func cancel() { }
```

Comment:
That's a great question, for the other 2 alerts we added a cancel action but not for the one that is only a message because there is nothing to cancel